### PR TITLE
ENH: Add lanczos window to scipy.signal.windows

### PR DESCRIPTION
**Pull request by serazing**
I have added the Lanczos window in `scipy.signal.windows` following the definition of other windows. I have added documentation, references and tests for this window. The reason why I added this window is because it is widely used among the climate research community for time and space filtering.

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @serazing, looks like a nice addition to me. There are a few PRs for new window functions open, so I'll send a message to the scipy-dev mailing list about those (that's where we decide on adding new functions).

File: scipy/signal/windows.py [hunk 1]
```diff
@@ -1658,6 +1736,7 @@ def exponential(M, center=None, tau=1., sym=True):
     ('hamming', 'hamm', 'ham'): (hamming, False),
     ('hanning', 'hann', 'han'): (hann, False),
     ('kaiser', 'ksr'): (kaiser, True),
+    ('lanczos', 'lcz'): (lanczos, True),
```
Comment (rgommers) [inline scipy/signal/windows.py]: I'd prefer to remove this. These additional spellings are present for backwards compatibility, but they're bad style so we shouldn't add new ones.

Comment (serazing) [inline scipy/signal/windows.py]: That's fair enough, I did not know it was for backwards compatibility. I have removed the spelling 'lcz' in the latest commit.

File: scipy/signal/tests/test_windows.py [hunk 2]
```diff
@@ -303,6 +304,23 @@ def test_basic(self):
                         [0, 0.25, 0.75, 1.0, 0.75, 0.25, 0])
 
 
+class TestLanczos(object):
+
+    def test_basic(self):
+        assert_allclose(signal.lanczos(6, False),
```
Comment (Eric89GXL) [inline scipy/signal/tests/test_windows.py]: Can you add a small comment about where these hard-coded values come from (hand calculations, Octave, etc.)?

**Review by larsoner — COMMENTED** [review]
Otherwise LGTM

**Comment by rgommers** [comment]
scipy-dev thread: https://mail.scipy.org/pipermail/scipy-dev/2017-January/021639.html

**Comment by larsoner** [comment]
Thread now lives at https://mail.python.org/pipermail/scipy-dev/2017-January/021654.html

@serazing any interest in reviving this with the newer `scipy/signal/windows/` structure?

**Comment by serazing** [comment]
I have to admit that I forgot to finalise this pull request. The newer `scipy/signal/windows/` structure is probably a good motivation to do it.

File: scipy/signal/windows/windows.py [hunk 3]
```diff
@@ -1970,6 +1970,78 @@ def dpss(M, NW, Kmax=None, sym=True, norm=None, return_ratios=False):
...
+           http://en.wikipedia.org/wiki/Window_function
+
+    Examples
+    --------
+    Plot the window and its frequency response:
+
+    >>> from scipy import signal
+    >>> from scipy.fftpack import fft, fftshift
+    >>> import matplotlib.pyplot as plt
+    >>> window = signal.lanczos(51)
```
Comment (pv) [inline scipy/signal/windows/windows.py]: `from scipy.signal.windows import lanczos`

File: scipy/signal/windows/windows.py [hunk 4]
```diff
@@ -1970,6 +1970,78 @@ def dpss(M, NW, Kmax=None, sym=True, norm=None, return_ratios=False):
...
+    w : ndarray
+        The window, with the maximum value normalized to 1 (though the value 1
+        does not appear if `M` is even and `sym` is True).
+
+    Notes
+    -----
+    The Lanczos window is defined as
+    .. math::  w(n) = sinc \left( \frac{2n}{M - 1} - 1 \right)
+    where
+    .. math::  sinc(x) = \frac{\sin(\pi x)}{\pi x}
```
Comment (pv) [inline scipy/signal/windows/windows.py]: Empty line before and after `.. math::` needed I guess.

**Review by pv — CHANGES_REQUESTED** [review]
Docstring needs updating vs. `scipy.signal.windows` but otherwise lgtm I think.

File: scipy/signal/windows/windows.py [hunk 5]
```diff
@@ -1970,6 +1970,78 @@ def dpss(M, NW, Kmax=None, sym=True, norm=None, return_ratios=False):
...
+           Physical Oceanography (Third Edition), Elsevier, pp 593-637.
+    .. [4] Wikipedia, "Window function",
+           http://en.wikipedia.org/wiki/Window_function
+
+    Examples
+    --------
+    Plot the window and its frequency response:
+
+    >>> from scipy import signal
+    >>> from scipy.fftpack import fft, fftshift
```
Comment (peterbell10) [inline scipy/signal/windows/windows.py]:
```suggestion
    >>> from scipy.fft import fft, fftshift
```

**Comment by AtsushiSakai** [comment]
I am trying to revive this PR in #16533.

**Comment by AtsushiSakai** [comment]
@kimstik Thank you. I agree it should be symmetric. I will try to fix it in #16533.

**Comment by AtsushiSakai** [comment]
This PR revived in #16533. So closing this. Thank you for your great PR. @serazing

**Pull request closed without merging**